Json.More/JsonSerializerOptionsExtensions.cs

```diff
@@ -60,11 +60,11 @@ public static JsonConverter<T> GetConverter<T>(this JsonSerializerOptions option
 	/// <param name="value">The value to serialize.</param>
 	/// <param name="typeInfo">An explicit typeInfo to use for looking up the Converter. If not provided, options.GetTypeInfo will be used.</param>
 	/// <returns>The value that was converted.</returns>
+	[UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "Deserialize is safe in AOT if the JsonSerializerOptions come from the source generator.")]
+	[UnconditionalSuppressMessage("AOT", "IL3050:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "Deserialize is safe in AOT if the JsonSerializerOptions come from the source generator.")]
```

Comment:
I'm not sure about this. Yes the suppression is correct if you are in fact passing options coming from a generated context, but suppressing these warnings in all cases means that callers wanting to use NativeAOT and not using a generated context will not get any warnings at all, but still crash at runtime. This explicitly goes against one of the design principles of the trim/AOT analyzers, we shouldn't hide warnings from users and rely on them doing the right thing (because they won't, and then they'll have a very hard time figuring out what's causing the problems).

Reply:
OK so yes the public methods should carry the RDC&RUC attributes but when we call them internally in a known safe way, we want those suppressed. Is there a good strategy for doing that?

Reply:
Can you declare internal proxy methods with those suppression attributes that just call those, and use the proxies?

Reply:
I see, so the red flag here is that we have a _public_ method annotated with these? But internal methods would be fine?

Reply:
Yup! Yeah if they're internal methods and _we_ know _we_ are calling them safely, then it's ok 🙂

Reply:
I fixed a couple of these, but mostly I just changed the justification message because we're doing the best we can to verify that the options set covers all the types we need. This is how System.Text.Json handles this scenario too.